Домашка. Week 2

anton0701 opened the pull request:
Сделал обязательную часть 
Из опционального сделал 2 пункта:
[опционально] Создать несколько конфигурационных файлов для разных сред для обоих проектов
[опционально] Написать парсер конфигов для обоих проектов

Постараюсь доделать опциональные пункты, пока не успеваю

Из того что не просили, но я сделал: поставил zap logger + добавил проверки полей для запросов и возвращаю ошибки, если проверки не проходят

P.S. postgres/cmd/query + postgres/cmd/query_with_squirel: файлы не используются -- просто повторял за Олегом, чтобы руку набить, вприницпе можно удалить

anton0701 committed:
db added
logger added
api handlers added

anton0701 committed:
SendMessage: db added + grpc-handler added

anton0701 committed:
добавлены проверки корректности полей в запросе

anton0701 committed:
local and prod configs added

anton0701 committed:
Unnecessary files deleted

anton0701 committed:
comments removed

anton0701 committed:
makefile: доавлены миграции для прод окружения

anton0701 committed:
unnessesary code removed

Shemistan commented on config/config.go:
```diff
@@ -0,0 +1,14 @@
+	"github.com/joho/godotenv"
+)
+
+func Load(path string) error {
```
Описание

anton0701 replied:
++

Shemistan commented on config/env/grpc.go:
```diff
@@ -0,0 +1,43 @@
+	grpcPortEnvName = "GRPC_PORT"
+)
+
+type GRPCConfig interface {
```
Описание

anton0701 replied:
++

Shemistan commented on config/env/grpc.go:
```diff
@@ -0,0 +1,43 @@
+	port string
+}
+
+func NewGRPCConfig() (GRPCConfig, error) {
```
Описание

anton0701 replied:
++

Shemistan commented on config/env/grpc.go:
```diff
@@ -0,0 +1,43 @@
+	}, nil
+}
+
+func (cfg *grpcConfig) Address() string {
```
Описание

anton0701 replied:
++

Shemistan commented on config/env/pg.go:
```diff
@@ -0,0 +1,33 @@
+	dsnEnvName = "PG_DSN"
+)
+
+type PGConfig interface {
```
Описание

anton0701 replied:
++

Shemistan commented on config/env/pg.go:
```diff
@@ -0,0 +1,33 @@
+	dsn string
+}
+
+func NewPGConfig() (PGConfig, error) {
```
Описание

anton0701 replied:
++

Shemistan commented on config/env/pg.go:
```diff
@@ -0,0 +1,33 @@
+	}, nil
+}
+
+func (cfg *pgConfig) DSN() string {
```
Описание

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -2,58 +2,285 @@ package main
+	return logger, nil
+}
+
+func (s *server) CreateChat(ctx context.Context, req *desc.CreateChatRequest) (*desc.CreateChatResponse, error) {
```
Описание

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -2,58 +2,285 @@ package main
+func (s *server) CreateChat(ctx context.Context, req *desc.CreateChatRequest) (*desc.CreateChatResponse, error) {
+	s.log.Info("Method Create-Chat", zap.Any("input params", req))
+
+	// TODO: текст ошибки вынести в константу?
```
пиcал в auth

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -2,58 +2,285 @@ package main
+
+	// TODO: текст ошибки вынести в константу?
+	// Проверка необходимых полей в запросе, User_IDs должен содеражть хотя бы 1 айди
+	if len(req.User_IDs) == 0 {
```
Лучше все эти проверки выносить в отдельную функцию validate() и там по мере необходимости добавлять параметры

anton0701 replied:
Создал интерфейс Validator с функцией Validate() - положил в /grpc/pkg/validator.go

Всем структурам запросов добавил метод Validate() с ресивером == ссылка на запрос. 
Имплементацию методов Validate() положил в файл /grpc/pkg/user_v1/user.go.

Если лучше это было организовать по-другому -- дай знать.

anton0701 replied:
++

Shemistan commented on grpc/grpc_server/main.go:
```diff
@@ -2,58 +2,285 @@ package main
+			PlaceholderFormat(sq.Dollar)
+
+		query, args, err := builderChatUsersInsert.ToSql()
+
```
Обычно хорошей практикой считается объединение вызова и обработки ошибки в один блок кода без пустой строки между ними, чтобы визуально подчеркнуть их связь.

anton0701 replied:
++

anton0701 committed:
добавлены описания методов

anton0701 committed:
добавлены описания публичных методов, интерфейсов и структур

anton0701 committed:
добавлены описания публичных методов, интерфейсов и структур

anton0701 committed:
добавлена валидация запросов:
- интерфейс Validator с функцией Validate()
- методы Validate(), где ресиверами являются структуры реквестов

anton0701 committed:
import minor
